main screen calculator setup

Shum616 opened the pull request.

Shum616 committed:
main screen calculator setup

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -48,11 +37,11 @@ class SharedIncomeViewModel @Inject constructor(
 
     fun calculateUnitedTaxUan(gross: Double, exchangeRate: Double): Double {
-        return Math.round(gross * exchangeRate * 0.05 * 100) / 100.0
+        return String.format("%.2f", Math.round(gross * exchangeRate * 0.05 * 100) / 100.0).toDouble()
```
Its very strange solution: round num in string format and then cast it to double. Previous solution with Math.round() was a lot better.

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -48,11 +37,11 @@ class SharedIncomeViewModel @Inject constructor(
 
     fun calculateUnidedSocialСontributionUan(gross: Double): Double {
-        return gross * 0.22
+        return String.format("%.2f", gross * 0.22).toDouble()
```
Its very strange solution: round num in string format and then cast it to double. Previous solution with Math.round() was a lot better.

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -64,7 +53,8 @@ class SharedIncomeViewModel @Inject constructor(
 
     fun calculateRemaining(gross: Double): Double {
-        return gross - calculateUnidedSocialСontributionUan(gross)
+        return String.format("%.2f", gross - calculateUnidedSocialСontributionUan(gross) -
```
Use Math.round(). Or even better: create extention for Double for rounding to 2 numbers

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -77,14 +67,16 @@ class SharedIncomeViewModel @Inject constructor(
     }
 
+    fun calculateTaxes(gross: Double) : Double{
+        return  String.format("%.2f", gross - calculateRemaining(gross)).toDouble()
```
Same comment as before

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/home/HomeFragment.kt:
```diff
@@ -22,19 +23,23 @@ class HomeFragment : BaseTabFragment() {
     ): View {
         binding = HomeFragmentLayoutBinding.inflate(inflater, container, false)
 
+        binding.cardVw1.setValues(getString(R.string.flat_rate_tax), "0")
```
Rename cardViews for its name to have more meaning

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/home/HomeFragment.kt:
```diff
@@ -22,19 +23,23 @@ class HomeFragment : BaseTabFragment() {
         binding.calculateBttn.setOnClickListener {
             val inputTxt = binding.txtField.text.toString()
-
+            val numToCalculate = inputTxt.toDouble()
```
If you have to parse string to get number it means that something is wrong with the logic. The viewModel should give view the "cooked" string or number without need for view to store it. The viewModel is a storage of transformed data itself

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/home/HomeFragment.kt:
```diff
@@ -22,19 +23,23 @@ class HomeFragment : BaseTabFragment() {
-                    binding.cardVw2.setValues(array[1].toString(), "income * 3")
-                    binding.cardVw3.setValues(array[2].toString(), "income * 4")
-                    binding.cardVw4.setValues(array[3].toString(), "income * 5")
+                    binding.cardVw1.setValues(getString(R.string.flat_rate_tax), viewModel.calculateUnitedTaxUan(numToCalculate, 1.0).toString())
```
In the current version you have string in viewModel that is parsed to double. Then this double number is passed to view, where it is being transformed into string once again. Remove unnecessary casting with logic review

Shum616 committed:
Merge branch 'refs/heads/develop' into main-screen-setup

# Conflicts:
#	app/src/main/res/layout/fragment_settings.xml
#	app/src/main/res/layout/home_fragment_layout.xml

Shum616 committed:
fixed issues

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -47,13 +36,13 @@ class SharedIncomeViewModel @Inject constructor(
+        (gross.toDouble() * exchangeRate * 0.05).roundToTwoDecimals().toString()
+
+
+    fun calculateUnidedSocialСontributionUan(gross: String) =
```
Why these function work with strings anyway? Isn't it more efficient to store everything in Double / Float and cast to strings before displaying only?

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/SharedIncomeViewModel.kt:
```diff
@@ -63,9 +52,12 @@ class SharedIncomeViewModel @Inject constructor(
-    fun calculateRemaining(gross: Double): Double {
-        return gross - calculateUnidedSocialСontributionUan(gross)
-    }
+    fun calculateRemaining(gross: String) =
```
Same here, why do gross value is coming as string?

lazyelf commented on app/src/main/java/com/example/businesstaxcalculator/ui/home/HomeFragment.kt:
```diff
@@ -22,19 +23,22 @@ class HomeFragment : BaseTabFragment() {
             val inputTxt = binding.txtField.text.toString()
-
-            val validRes = viewModel.incomeValidation(inputTxt)
+            val validRes = viewModel.incomeValidation(binding.txtField.text.toString())
```
Its best to extract here a numerable, not string

Shum616 replied:
income validator set two values to validRes properties: Boolean for isEmpty check and Int for input value check. So there is no way to get numerable for viewmodel methods. That`s why inputTxt is.

viktoriiamiketetalab committed:
main screen setup

- changed stored viewModel types to double from string
- replaced 'calculate' for 'get' in fun names
- changed project structure
- fixed validator to validate fractional
